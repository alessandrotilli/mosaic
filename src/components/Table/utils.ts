export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
