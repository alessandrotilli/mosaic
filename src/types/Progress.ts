import { IBase } from "./Base";

type ProgressType = "circular" | "linear";

export type ProgressProps = IBase & {
  type: ProgressType;
  withLabel?: boolean;
  color?: "primary" | "success" | "error";
  variant?: "determinate" | "indeterminate";
  value?: number;
};
