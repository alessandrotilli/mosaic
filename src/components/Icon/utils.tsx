import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
