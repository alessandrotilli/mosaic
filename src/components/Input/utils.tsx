import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
