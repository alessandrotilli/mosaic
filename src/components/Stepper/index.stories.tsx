import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { configure, expect, fireEvent, fn, within } from "@storybook/test";

import { logInfo } from "../../utils/logger";
import getDocsPage from "../../utils/stories";
import LocaleDecorator from "../../utils/stories/decorators/Locale";
import StepperDecorator from "../../utils/stories/decorators/Stepper";
import Progress from "../Progress";

import { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, localizedStepper } from ".";

const COMPONENT_NAME = "Stepper";
localizedStepper.displayName = COMPONENT_NAME;
//ProgressWithProps.displayName = COMPONENT_NAME;

configure({ testIdAttribute: "data-cy" });

const meta = {
  title: "Navigation/Stepper",
  component: localizedStepper,
  decorators: [StepperDecorator, LocaleDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Stepper Component",
          url: "https://mui.com/material-ui/react-stepper/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
    },
  },
} satisfies Meta<typeof localizedStepper>;

export default meta;
type Story = StoryObj<typeof meta>;

const onClickMockBack = fn(() => logInfo(COMPONENT_NAME, "onClick handler"));
const onClickMockNext = fn(() => logInfo(COMPONENT_NAME, "onClick handler"));
const onClickMockFinish = fn(() => logInfo(COMPONENT_NAME, "onClick handler"));

export const Primary: Story = {
  args: {
    activeStep: 1,
    finishContent: "All Steps are completed",
    onBackClick: onClickMockBack,
    onNextClick: onClickMockNext,
    onFinishClick: onClickMockFinish,
    stepsList: [
      { label: "STEP1", content: "CONTENT PAGE 1" },
      { label: "STEP2", content: "CONTENT PAGE 2" },
      { label: "STEP3", content: <Progress type="linear" /> },
    ],
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const buttonBack = canvas.getByTestId(`${DATA_CY_DEFAULT}-back`);
    const buttonNext = canvas.getByTestId(`${DATA_CY_DEFAULT}-next`);
    if (!buttonBack) {
      return;
    }

    fireEvent.click(buttonBack);
    await expect(onClickMockBack).toHaveBeenCalledTimes(onClickMockBack.mock.calls.length);
    fireEvent.click(buttonNext);
    fireEvent.click(buttonNext);
    await expect(onClickMockNext).toHaveBeenCalledTimes(onClickMockNext.mock.calls.length);
  },
};

export const ActiveStep: Story = {
  args: {
    ...Primary.args,
    activeStep: 3,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const buttonFinish = canvas.getByTestId(`${DATA_CY_DEFAULT}-finish`);
    fireEvent.click(buttonFinish);
    await expect(onClickMockFinish).toHaveBeenCalledTimes(onClickMockFinish.mock.calls.length);
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    labelBackButton: "locale.back",
    labelNextButton: "locale.next",
    labelFinishButton: "locale.finish",
  },
};
