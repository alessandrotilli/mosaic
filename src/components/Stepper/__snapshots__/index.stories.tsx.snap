// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/Stepper ActiveStep smoke-test 1`] = `
<div>
  <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1et7qun-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-10mg1vw-MuiStepper-root"
           data-cy="Stepper"
      >
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-n7tliy-MuiStep-root">
          <span class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-completed MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
                   data-testid="CheckCircleIcon"
              >
                <path d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z">
                </path>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-completed MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP1
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-completed css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-completed MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
                   data-testid="CheckCircleIcon"
              >
                <path d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z">
                </path>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-completed MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP2
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-active css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  3
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP3
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        <div class="MuiBox-root css-70qvj9">
          <div class="MuiBox-root css-1bhivts">
            <span class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
                  role="progressbar"
                  data-cy="progress"
            >
              <span class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1">
              </span>
              <span class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2">
              </span>
            </span>
          </div>
        </div>
      </div>
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        <div style="display: flex; justify-content: space-between;">
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  data-cy="button"
                  style="margin-right: 1px;"
          >
            Back
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </button>
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  data-cy="button"
          >
            Next
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navigation/Stepper Localized smoke-test 1`] = `
<div>
  <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1et7qun-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-10mg1vw-MuiStepper-root"
           data-cy="Stepper"
      >
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <span class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  1
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                locale.step1
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  2
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                locale.step2
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  3
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                locale.step3
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        CONTENT PAGE 1
      </div>
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        <div style="display: flex; justify-content: space-between;">
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="-1"
                  type="button"
                  disabled
                  data-cy="button"
                  style="margin-right: 1px;"
          >
            Dietro
          </button>
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  data-cy="button"
          >
            Avanti
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navigation/Stepper Primary smoke-test 1`] = `
<div>
  <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1et7qun-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-10mg1vw-MuiStepper-root"
           data-cy="Stepper"
      >
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <span class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  1
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP1
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  2
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP2
              </span>
            </span>
          </span>
        </div>
        <div class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root">
          <div class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root">
            <span class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line">
            </span>
          </div>
          <span class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root">
            <span class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer">
              <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                   focusable="false"
                   aria-hidden="true"
                   viewbox="0 0 24 24"
              >
                <circle cx="12"
                        cy="12"
                        r="12"
                >
                </circle>
                <text class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      x="12"
                      y="12"
                      text-anchor="middle"
                      dominant-baseline="central"
                >
                  3
                </text>
              </svg>
            </span>
            <span class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer">
              <span class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label">
                STEP3
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        CONTENT PAGE 1
      </div>
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
        <div style="display: flex; justify-content: space-between;">
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="-1"
                  type="button"
                  disabled
                  data-cy="button"
                  style="margin-right: 1px;"
          >
            Back
          </button>
          <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1p5nen7-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                  data-cy="button"
          >
            Next
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
